ztp/remote-worker-day0/ztp-ai/README.md

```diff
@@ -151,4 +151,163 @@ While this approach was tested and worked for UPI deployments, Assisted Installe
 
 Now that we have the prereqs and patches ready we can run the cluster deployment, we just need to create all objects inside the [assets/ztp-cluster/](./assets/ztp-cluster/) in our Hub cluster.
 
-You can see the deployment in action [here](https://drive.google.com/file/d/1yDYMsawPBNgjM3zm9-7ZqSHuiUBrOk2S/view?usp=sharing).
\ No newline at end of file
+You can see the deployment in action [here](https://drive.google.com/file/d/1yDYMsawPBNgjM3zm9-7ZqSHuiUBrOk2S/view?usp=sharing).
+
+### **Ingress and Egress on the Remote Worker Node**
+
+**Ingress**
+
+* OpenShift Router/s running on the master/non-remote worker node/s:
+  
+  * Ingress to applications running on the remote worker node works just fine. This is expected since the routers are running on the master nodes and from there connection to the remote worker node will happen at SDN level.
+
+* OpenShift Router/s running on the remote worker node/s:
+
+  * Default ingress controller on remote worker node/s:
+
+    * The remote node doesn't have an interface on the network where the IngressVIP is located, and as such, if you try to move ingress there the router pod will start and br-ex will configure the IngressVIP, but ingress will be broken since traffic cannot be properly routed.
+
+  * Sharded ingress on remote node:
+
+    * We configured a sharded ingress controller ([official docs](https://docs.openshift.com/container-platform/4.9/networking/ingress-operator.html#nw-ingress-sharding_configuring-ingress)). The configuration for this ingress controller looks like this:
+
+      ~~~yaml
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: rwn
+        namespace: openshift-ingress-operator
+      spec:
+        replicas: 1
+        domain: apps-rwn.ztp.e2e.bos.redhat.com
+        nodePlacement:
+          nodeSelector:
+            matchLabels:
+              kubernetes.io/hostname: "openshift-worker-2.ztp.e2e.bos.redhat.com"
+        routeSelector:
+          matchLabels:
+            type: rwn-route
+      ~~~
+
+    * Since we need to route traffic to our new domain `apps-rwn.ztp.e2e.bos.redhat.com`, we deployed an external HAProxy that redirects connections to the remote worker node/s where the routers for this domain are running.
+
+    * We labeled our OpenShift Routes targetting apps running on remote worker nodes with the label `type: rwn-route` and then the traffic was handled by the OpenShift router running on the remote worker node and the app was accessed successfully.
+
+**Egress**
+
+* **Egress IP from non-RWN machineNetwork for apps running on RWN:**
+
```

Comment:
A small network diagram here would be nice :)

Reply:
Added diagrams for egress and also ingress scenarios.